### PR TITLE
chore(extension): AR button

### PR DESCRIPTION
**Pull request by jashanbhullar**
- - Added the icon
- addded a todo
- - tried to get all the layers but shows undefined

### CODE REVIEW

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx [hunk 1]
```diff
@@ -19,9 +21,23 @@ export const AppHeader: FC = () => {
   const [showShareModal, setShowShareModal] = useAtom(showShareModalAtom);
   const theme = useTheme();
   const isMobile = useMediaQuery(theme.breakpoints.down("mobile"));
+  const getSharedRootLayers = useSetAtom(getSharedRootLayersAtom);
```
Comment (keiya01) [inline extension/src/prototypes/view/ui-containers/AppHeader.tsx:24]:
```suggestion
  const rootLayers = useAtomValue(rootLayersAtom);
```

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx [hunk 2]
```diff
@@ -38,6 +59,13 @@ export const AppHeader: FC = () => {
       <IconButton>
         <PaperPlaneTilt onClick={() => setShowShareModal(true)} />
       </IconButton>
+      {isMobile && arSharedLayers && (
```
Comment (keiya01) [inline extension/src/prototypes/view/ui-containers/AppHeader.tsx:62]: What is arSharedLayers for?

```suggestion
      {isMobile && (
```

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx [hunk 3]
```diff
@@ -14,14 +15,34 @@ import { MainMenuButton } from "./MainMenuButton";
...
+  const getRootLayers = useAtomValue(rootLayersAtom);
+  const [arSharedLayers, setArSharedLayers] = useState<string | undefined>();
+
+  useEffect(() => {
+    (async () => {
+      if (!getRootLayers || getRootLayers?.length === 0 || !arURL) return;
+      const layers = getRootLayers.map(({ type, id }) => ({
+        type,
+        id,
+      }));
```
Comment (keiya01) [inline extension/src/prototypes/view/ui-containers/AppHeader.tsx:36]: We should pass the `id`, `currentDataIdAtom` and type. So we need to make an atom to get the currentDataIdAtom.
```
const rootLayersForAR = useMemo(() => atom((get) => {
  const rootLayers = get(rootLayersAtom);
  return rootLayers.map(({ type, id, currentDataIdAtom }) => ({
    type,
    datasetId: id,
    dataId: get(currentDataIdAtom),
  }))
}), []);
```

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx [hunk 4]
```diff
@@ -38,6 +59,13 @@ export const AppHeader: FC = () => {
       <IconButton>
         <PaperPlaneTilt onClick={() => setShowShareModal(true)} />
       </IconButton>
+      {isMobile && arSharedLayers && (
+        <IconButton
+          size="small"
+          onClick={() => window.open(arSharedLayers, "_blank", "noopener,noreferrer")}>
```
Comment (keiya01) [inline extension/src/prototypes/view/ui-containers/AppHeader.tsx:65]:
```suggestion
          href={arSharedLayers}
          target="_blank">
```

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx (hunk 3 above)
Comment (jashanbhullar) [reply extension/src/prototypes/view/ui-containers/AppHeader.tsx]: Thanks for this. Fixed :heart:

File: extension/src/prototypes/view/ui-containers/AppHeader.tsx (hunk 4 above)
Comment (jashanbhullar) [reply extension/src/prototypes/view/ui-containers/AppHeader.tsx]: Fixed

**Review by keiya01 — APPROVED** [review]

**Pull request merged**